bind/motor-wasm/wasm.go

```
@@ -1,18 +1,263 @@
+//go:build wasm
 // +build wasm
 
 package main
 
 import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"syscall/js"
+
 	"github.com/sonr-io/sonr/pkg/crypto"
+	"github.com/sonr-io/sonr/pkg/did"
+	"github.com/sonr-io/sonr/pkg/did/ssi"
 )
 
-func main() {
-	_, err := crypto.GenerateWallet()
-	check(err)
+var (
+	errWalletExists    = errors.New("mpc wallet already exists")
+	errWalletNotExists = errors.New("mpc wallet does not exist")
+)
+
+type motor struct {
+	//	node *host.SonrHost
+	wallet *crypto.MPCWallet
+	// host   host.SonrHost
+}
+
+var instance *motor
+
+// NewWallet creates a new mpc based wallet.
+func NewWallet() error {
+	if instance != nil {
+		return errWalletExists
+	}
+	w, err := crypto.GenerateWallet()
+	if err != nil {
+		return err
+	}
+	instance = &motor{
+		wallet: w,
+	}
+	return nil
+}
+
+// Address returns the address of the wallet.
+func Address() string {
+	if instance == nil {
+		return ""
+	}
+	addr, err := instance.wallet.Address()
+	if err != nil {
+		return ""
```

Should we be returning the errors in some way instead of swallowing them

---

+1, the binding should support errors. If wasm only supports single return values, we could return arrays that represent tuples.

---

good call